bloggies_backend/src/models/user.ts

```diff
@@ -76,6 +77,49 @@ export default class User {
     return res.rows[0];
   }
 
+  /** Update an existing user */
+  static async updateUser(id: number, updateData: any) {
+    try {
+      let query = "";
+
+      for (let key in updateData) {
+        query = query + ` ${key} = '${updateData[key]}', `;
+      }
+
+      await db.query(
+        `UPDATE users
+        SET ${query}  
+        WHERE user_id = $1`,
+        [id]);
+    } catch (err) {
+      throw new ExpressError(`Err: ${err}`, 400);
+    }
+  }
+
+  /** Sets membership_status to INACTIVE. Sets membership_end_date 
+   * to CURRENT_TIMESTAMP. via subscription id */
+  static async cancelSubscription(subscriptionId: string) {
```

Comment:
See my comment below - similar case here - we can use the same function and send it "inactive".

Reply:
For this one, I opted for making a new method for `cancelSubscription` because we wanted to keep the `membership_start_date` and use CURRENT_TIMESTAMP for the `membership_end_date` instead of `NULL`. This was preferred because it could help us build some UI that can let the user know about their last subscription duration, if we decide to!